### PR TITLE
[OMNIML-4998] Per-expert weight quantization on TEGroupedLinear

### PR DESCRIPTION
**Pull request by hychiang-git**
### What does this PR do?

Type of change: New feature

Per-expert (axis=0) weight quantization on `_QuantTEGroupedLinear`, plus the cluster infrastructure to verify it. Consolidates three Jira tickets onto one branch:

- **OMNIML-4998** — Relax `_QuantMegatronTEGroupedLinear._process_quantizer_amax`'s per-tensor restriction. Per-expert `_amax` of shape `[num_gemms, 1, 1]` is calibrated via `iter_weights_for_calibration`'s stacked weight, broadcast through the existing axis-N machinery in `TensorQuantizer.forward`, and round-tripped via `sharded_state_dict` save/restore. Per-tensor (`axis=None`) path remains bit-for-bit equivalent.
- **OMNIML-5029** — `megatron_install_path` field on `SlurmConfig` / `aws_cmh_slurm_factory`. Bind-mounts the packaged Megatron-LM so `make_tp_sharded_tensor_for_checkpoint(..., allow_shape_mismatch=True)` resolves against `nemo:25.11.nemotron_3_nano` (previously deadlocked the dist-checkpoint metadata exchange for 10 minutes).
- **OMNIML-5030** — Test fix: enable `sequence_parallel=True` in the MoE+TP `test_moe_sharded_state_dict` config so the TEGrouped path doesn't trip the post-OMNIML-5029 assertion.

### Usage

```python
import modelopt.torch.quantization as mtq

config = {
    "algorithm": "max",
    "quant_cfg": [
        {"quantizer_name": "*", "enable": False},
        {
            "quantizer_name": "*experts.linear_fc*.weight_quantizer",
            "cfg": {"num_bits": 8, "axis": 0, "enable": True},
        },
    ],
}
model = mtq.quantize(model, config, forward)
```

### Testing

`tests/gpu_megatron/torch/quantization/plugins/test_megatron.py::test_te_grouped_per_expert_sharded_state_dict` (new) — per-expert axis=0 calibration + dist-checkpoint save/restore round-trip on `tp=2 ep=2 etp=1 num_moe_experts=4`. **PASSED 4/4 ranks on aws-cmh** (SLURM 511920, experiment `cicd_1781150929`, container `nvcr.io/nvidia/nemo:25.11.nemotron_3_nano`).

Existing tests continue to pass:
- `test_moe_sharded_state_dict[*-True]` with the OMNIML-5030 `sequence_parallel` fix.
- Per-tensor (`axis=None`) paths — bit-for-bit equivalent (no behavior change).

Cluster yaml: `services/model-optimizer/test/omniml_4998_per_expert_amax_smoke.yaml` on nmm-sandbox `hungyuehc/omniml-5029 @ 24860c3`.

### Before your PR is "*Ready for review*"

- Is this change backward compatible?: ✅ — per-tensor (`axis=None`) path is bit-for-bit equivalent; per-expert is a new opt-in via existing axis-N machinery, not a behavior change for existing users.
- New PIP dependency / copied code?: N/A.
- Did you write any new necessary tests?: ✅ — `test_te_grouped_per_expert_sharded_state_dict`.
- Did you update Changelog?: ❌ — TBD; will add a changelog entry before marking Ready-for-review.
- Did you get Claude approval on this PR?: N/A — will run `/claude review` before marking Ready-for-review.

### Additional Information

**Related Jira:**
- OMNIML-4998 (primary scope; AC-1 and AC-2 met)
- OMNIML-5029 (launcher fix)
- OMNIML-5030 (test fix)
- OMNIML-5051 (follow-up: per-expert quant + TE-grouped MoE LoRA compose)

**Relationship to PR #1550** (`jennifchen/te_per_expert`, WIP): PR #1550 explores the same goal with a different design — opt-in env-var `MODELOPT_TEGROUPED_PER_EXPERT_QUANTIZER=1` to create N separate `weight_quantizer_i` instances. This PR uses the single-quantizer + axis=0 design instead, which reuses modelopt's existing axis-N machinery without a new env-var or per-gemm state. The two designs serve overlapping but not identical needs:

- This PR covers "all experts share the same quantizer config but each has its own amax" — common case.
- PR #1550 covers "each expert independently configurable" — rarer; can be layered later.

PR #1550 is currently WIP and CONFLICTING vs main. Open to discussion on whether to merge approaches; this PR is the focused common-case implementation.

## Summary by CodeRabbit

* **New Features**
  * Added per-expert weight quantization support for distributed checkpointing in MoE models.
  * Integrated Megatron-LM mounting into launcher tools for both Slurm and Docker environments.

* **Improvements**
  * Enhanced tensor-parallel and expert-parallel weight quantization synchronization during calibration and checkpointing.

* **Tests**
  * Added regression test for per-expert weight quantization across distributed configurations.
  * Expanded launcher integration test coverage for Docker and Slurm environments.

### CODE REVIEW

**Comment by hychiang-git** [comment]
Related to PR #1550 — same scope (per-expert weight quantization on TEGroupedLinear), alternative design:

- This PR (#1671): single `weight_quantizer` slot with `axis=0`, `_amax` shape `[num_gemms, 1, 1]`. Per-tensor (`axis=None`) path remains bit-for-bit equivalent. AC-1 and AC-2 GPU-verified end-to-end on aws-cmh.
- PR #1550 (`jennifchen/te_per_expert`): opt-in env var creating N separate `weight_quantizer_i` instances, one per gemm.

Happy to discuss merging approaches. Cross-linking so reviewers on either PR have visibility into the other.

**Comment by hychiang-git** [comment]
We will use N-quantizers implementation because it would be more flexible in the Sparse Model and future features like MoE pruning.
